### ISSUE TITLE
Breaks Magento compilation without Composer development dependencies

### ISSUE THREAD

**Issue by JosephLeedy** [flancer32/mage2_ext_csp#1]
# Expected Result
`bin/magento setup:di:compile` command should execute without errors.

# Actual Result
The following error is thrown:
```
Fatal error: Uncaught Error: Class 'PHPUnit\Framework\TestCase' not found in /opt/atlassian/pipelines/agent/build/vendor/flancer32/mage2_ext_csp/test/manual/Model/Collector/DbTest.php:12
Stack trace:
#0 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(135): require_once()
#1 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(117): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->includeClasses(Array, '/opt/atlassian/...')
#2 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(88): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->extract(Object(RecursiveIteratorIterator))
#3 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/App/Task/Operation/RepositoryGenerator.php(61): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->getList('/opt/atlassian/...')
#4 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/App/Task/Manager.php( in /opt/atlassian/pipelines/agent/build/vendor/flancer32/mage2_ext_csp/test/manual/Model/Collector/DbTest.php on line 12
Repositories code generation... 1/8 [==>-----------------]  12% < 1 sec 95.0 MiBPHP Fatal error:  Uncaught Error: Class 'PHPUnit\Framework\TestCase' not found in /opt/atlassian/pipelines/agent/build/vendor/flancer32/mage2_ext_csp/test/manual/Model/Collector/DbTest.php:12
Stack trace:
#0 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(135): require_once()
#1 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(117): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->includeClasses(Array, '/opt/atlassian/...')
#2 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(88): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->extract(Object(RecursiveIteratorIterator))
#3 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/App/Task/Operation/RepositoryGenerator.php(61): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->getList('/opt/atlassian/...')
#4 /opt/atlassian/pipelines/agent/build/setup/src/Magento/Setup/Module/Di/App/Task/Manager.php( in /opt/atlassian/pipelines/agent/build/vendor/flancer32/mage2_ext_csp/test/manual/Model/Collector/DbTest.php on line 12
```

# Steps to Reproduce
1. Install Magento with `composer install --no-dev`
2. Install extension via Composer
3. Dump optimized autoloader without development dependencies by running `composer dump-autoload -o --no-dev`
4. Run `bin/magento setup:di:compile`

# Environment
1. Magento 2.3.5-p1
2. PHP 7.3.17

# Possible Cause
Magento's `composer.json` file contains the following statement:
```json
    "autoload": {
        "exclude-from-classmap": [
            "**/dev/**",
            "**/update/**",
            "**/Test/**",
         ]
   }
```
This prevents files in specific directories from being added to Composer's `autoload.php` files.

# Possible Solution
Rename the `test/` directory to `Test/`.

**Comment by flancer64** [comment]
Hi, @JosephLeedy !

Thanks for the issue. I've tried to reproduce situation but I had the following error:

```
$ bin/magento setup:di:compile
Warning: include(/.../vendor/composer/../../generated/code/Magento/Framework/App/ResourceConnection/Proxy.php): failed to open stream: No such file or directory in /.../vendor/composer/ClassLoader.php on line 444
```

By the way, I understand possible reason for the error you mentioned and I'll rename `./test/` to `./dev/`. Because scripts in module's `./test/` folder are not real tests but is development environment.

**Comment by flancer64** [comment]
I've created new branch named `release` and remove `./dev/` & ./docs/` folders from this branch. New release 0.0.4 is created based on this branch. I hope `Class 'PHPUnit\Framework\TestCase' not found` error will not be appeared more.

Thanks again and have a good day!

**Comment by JosephLeedy** [comment]
Thank you for the prompt reply and resolution @flancer64! I can confirm that the issue is resolved with the change in version 0.0.4.

*Note*: to fix the error regarding the missing `Proxy` class, you need to clear out `generated/` (`rm -rf generated/*`) before running `composer dump-autoload`.
